src/libsync/pushnotifications.h

```diff
@@ -0,0 +1,45 @@
+#ifndef PUSHNOTIFICATIONS_H
+#define PUSHNOTIFICATIONS_H
+
+#include "websocket.h"
+
+namespace OCC {
+
+class Account;
+class AbstractCredentials;
+
+class PushNotifications : public QObject
+{
+    Q_OBJECT
+
+public:
+    // Question: Is it okay to pass a raw account pointer? Because PushNotifications don't really own the account. The PushNotifications
+    // object is owned by the account state. Why not passing AccountState? Maybe would be better but this makes testing more difficult.
+    PushNotifications(Account *account, QSharedPointer<AbstractWebSocket> webSocket = QSharedPointer<AbstractWebSocket>(new WebSocket));
```

Comment:
I wouldn't receive the websocket as a parameter, it should be built internally in my opinion.

Also please make the ctor explicit and add the `QObject *parent = nullptr` as last parameter so that it can be given a proper parent at build time (typically the AccountState which will create it).

Reply:
Okay but then it is difficult to stub the websocket in tests? We could make the test class a friend and let it alter the webSocket variable or as you said just test with a mock server.

Reply:
Yes, I'd go for the fake server.